time record_duration
time video_length
uint32 fps
uint32 angle
bool direction